src/Credentials/GCECredentials.php

```diff
@@ -448,14 +449,20 @@ public function getCacheKey()
     }
 
     /**
-     * @return array{access_token:string,expires_at:int}|null
+     * @return array<mixed>|null
      */
     public function getLastReceivedToken()
     {
         if ($this->lastReceivedToken) {
+            if (array_key_exists('id_token', $this->lastReceivedToken)) {
+                return [
+                    'id_token' => $this->lastReceivedToken['id_token']
+                ];
```

Comment:
Done

Reply:
Oh that's strange. Maybe it was to avoid including the `expires_in` field of the cached token. If that's the case, then we should just unset it _before_ caching. To me it's odd that we look at the cached token at all instead of just returning it.

Reply:
Looking at the code, there _is_ an `expires_in` field being returned from the metadata server, so that's my guess as to why it's being done there:

https://github.com/googleapis/google-auth-library-php/pull/510/files#diff-3b0613528ba5198b6170cd802316c6b3926a6f635f05aa9cf262f03f97102cb8R434

Reply:
Makes sense.